src/libpmemfile/preload.c

```diff
@@ -60,6 +60,29 @@
 #include "libpmemfile-core.h"
 #include "util.h"
 
+/*
+ * Some syscalls that are missing on older kernels.
+ */
+#ifndef SYS_renameat2
+#define SYS_renameat2 316
+#endif
+
+#ifndef SYS_execveat
+#define execveat 322
+#endif
+
+#ifndef SYS_copy_file_range
+#define SYS_copy_file_range 326
+#endif
+
+#ifndef SYS_preadv2
+#define SYS_preadv2 327
+#endif
+
+#ifndef SYS_pwritev2
+#define SYS_pwritev2 328
+#endif
```

Comment:
I see a lot that is x86-64 specific. The list of syscalls we handle, the six long arguments - expected to be 64 bits. To be general, it would need some general table of syscalls, and non-hardwired argument type, probably non-hardwired argument-count, and who knows what.